### PR TITLE
db-cleaning script - minimal update #6460

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: apps/backend-ops/src/db-cleaning.spec.ts [hunk 1]
```diff
@@ -89,26 +93,29 @@ describe('DB cleaning script', () => {
...
     ];
 
     // Add a new auth user that is not on db
-    const fakeAuthUser = { uid: 'D', email: 'D@fake.com' };
-
+    const fakeAuthUser = { uid: 'D', email: 'D@fake.com', providerData: [{ uid: 'D', email: 'D@fake.com', providerId: 'password' }] };
     adminAuth.users.push(fakeAuthUser);
 
+    // Add an anonymous user that should not be deleted even if not on DB
+    const anonymousUser = { uid: 'anonymous', providerData: [] };
```
Comment (RemcoSimonides) [inline apps/backend-ops/src/db-cleaning.spec.ts:106]: No need to add providerData? `{ uid: 'anonymous', providerId: 'anonymous' }`

File: apps/backend-ops/src/users.ts [hunk 2]
```diff
@@ -62,7 +62,8 @@ export async function removeUnexpectedUsers(expectedUsers: UserConfig[], auth: A
   do {
     const result = await auth.listUsers(1000, pageToken);
 
-    const users = result.users;
+    // Anonymous users filtering
+    const users = result.users.filter(u => u.providerData.length !== 0);
```
Comment (RemcoSimonides) [inline apps/backend-ops/src/users.ts:66]: and if you provide providerId in earlier comment, then filter on that instead of array length

**Review by RemcoSimonides — APPROVED** [review]

File: apps/backend-ops/src/db-cleaning.spec.ts (hunk 1 above)
Comment (phpgeekfr) [reply apps/backend-ops/src/db-cleaning.spec.ts:106]: no, when user is logged in as anonymous, `providerData` is an empty array

File: apps/backend-ops/src/users.ts (hunk 2 above)
Comment (phpgeekfr) [reply apps/backend-ops/src/users.ts:66]: No, we need to test what a real anonymous user will have in `providerData` array. If anonymous , this array is empty 
see https://stackoverflow.com/questions/39640574/how-to-bulk-delete-firebase-anonymous-users

**Pull request merged**